From 229929d09eb6ef2a0611eeef31e51fd174c8d4a7 Mon Sep 17 00:00:00 2001
From: Evan Huus <eapache@gmail.com>
Date: Tue, 19 Mar 2013 15:51:19 +0000
Subject: [PATCH 7/7] Change some ints to guints (as they already are in
 trunk) so that negative values don't falsely pass the
 bounds checks and cause a crash.

Fixes https://bugs.wireshark.org/bugzilla/show_bug.cgi?id=8499

svn path=/trunk-1.8/; revision=48419
---
 epan/dissectors/packet-websocket.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/epan/dissectors/packet-websocket.c b/epan/dissectors/packet-websocket.c
index 01bbe24..0736b47 100644
--- a/epan/dissectors/packet-websocket.c
+++ b/epan/dissectors/packet-websocket.c
@@ -117,14 +117,14 @@ static const value_string ws_close_status_code_vals[] = {
 
 #define MAX_UNMASKED_LEN (1024 * 64)
 tvbuff_t *
-tvb_unmasked(tvbuff_t *tvb, const int offset, int payload_length, const guint8 *masking_key)
+tvb_unmasked(tvbuff_t *tvb, const guint offset, guint payload_length, const guint8 *masking_key)
 {
 
   gchar *data_unmask;
   tvbuff_t  *tvb_unmask    = NULL;
-  int i;
+  guint i;
   const guint8 *data_mask;
-  int unmasked_length = payload_length > MAX_UNMASKED_LEN ? MAX_UNMASKED_LEN : payload_length;
+  guint unmasked_length = payload_length > MAX_UNMASKED_LEN ? MAX_UNMASKED_LEN : payload_length;
 
   data_unmask = g_malloc(unmasked_length);
   data_mask = tvb_get_ptr(tvb, offset, unmasked_length);
-- 
1.7.10.4

